tools/template-check/cmd/unusedtmpl.go

```diff
@@ -0,0 +1,199 @@
+	Examples []struct {
+		Name string
+	}
+}
+
+func newUnusedTmplCmd(rootOptions *rootOptions) *cobra.Command {
+	o := &unusedTmplOptions{
+		rootOptions: rootOptions,
+		stdout:      os.Stdout,
+	}
+	command := &cobra.Command{
+		Use:   "unused-tmpl",
+		Short: unusedTmplDesc,
+		Long:  unusedTmplDesc,
+		RunE: func(c *cobra.Command, args []string) error {
+			return o.run()
+		},
+	}
+	command.Flags().StringSliceVar(&o.fileList, "file-list", []string{}, "file list to check")
+	return command
+}
+
+func (o *unusedTmplOptions) run() error {
+	if len(o.fileList) == 0 {
+		return nil
+	}
+	newCustomTmpls, newExamples := processInputFiles(o.fileList)
+
+	found := false
+	// get repo dir from tmpl files
+	repoPath := strings.Split(o.fileList[0], "/mmv1/")[0]
+	dir := filepath.Join(repoPath, "mmv1", "products")
+
+	productFiles, err := yamlFiles(dir)
+	if err != nil {
+		return err
+	}
+	if len(newCustomTmpls) > 0 {
+		customTempls, err := findTmpls(productFiles)
+		if err != nil {
+			return err
+		}
+		for _, file := range newCustomTmpls {
+			templatePath := strings.ReplaceAll(file, repoPath+"/mmv1/", "")
+			if _, ok := customTempls[templatePath]; !ok {
+				found = true
+				fmt.Fprintf(os.Stderr, "File %s not used in any product yaml.\n", file)
+			}
+		}
+	}
+	if len(newExamples) > 0 {
+		ex, err := findExamples(productFiles)
+		if err != nil {
+			return err
+		}
+		for _, file := range newExamples {
+			baseName := filepath.Base(file)
+			newExName := strings.TrimSuffix(baseName, ".tf.tmpl")
+			if _, ok := ex[newExName]; !ok {
+				found = true
+				fmt.Fprintf(os.Stderr, "File %s not used in any product yaml.\n", file)
+			}
+		}
+
+	}
+	if found {
+		return fmt.Errorf("found templates not used")
+	}
+	return nil
+}
+
+func processInputFiles(fileList []string) (customTmpls []string, examples []string) {
+	for _, v := range fileList {
+		if exampleFilePathReg.MatchString(v) {
+			examples = append(examples, v)
+		} else if strings.Contains(v, "mmv1/templates/terraform") && strings.HasSuffix(v, ".tmpl") {
+			customTmpls = append(customTmpls, v)
+		} else {
+			fmt.Printf("Skipping check for file %s\n", v)
+		}
+	}
+	return
+}
+
+func (t *tree) walkTree(tree map[any]any) {
+	for _, value := range tree {
+		switch v := value.(type) {
+		case []any:
+			for _, v1 := range v {
+				if val, ok := v1.(map[any]any); ok {
+					t.walkTree(val)
+				}
+			}
+		case map[any]any:
+			t.walkTree(v)
+		case string:
+			if strings.HasSuffix(v, ".tmpl") {
+				t.tmplPaths[v] = true
+			}
+		default:
+		}
+	}
+}
+
+func yamlFiles(dir string) ([]string, error) {
+	var allYamlFiles []string
+	err := filepath.Walk(dir, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if filepath.Ext(path) == ".yaml" {
+			allYamlFiles = append(allYamlFiles, path)
+		}
+		return nil
+	})
+	if err != nil {
+		return nil, err
+	}
+	return allYamlFiles, nil
+}
+
+func findTmpls(yamlFiles []string) (map[string]bool, error) {
```

Comment:
If possible, consider adding a one-sentence comment to explain some of the functions. Non-blocker though, thanks!